### ISSUE TITLE
Create new `TagOverflow`component

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#4086]

**Comment by elycheea** [comment]
@szinta This initial implementation leverages `TagSet`. I _think_ we discussed opening a followup ticket to remove the dependency on the original version so we can deprecate it in the future. Is that correct? Want to double check before I add that as another issue.

**Comment by szinta** [comment]
That's right @elycheea. We need one new ticket to remove the dependency on Tagset component.
